cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/GoalOptimizer.java

```diff
@@ -121,7 +124,9 @@ public GoalOptimizer(KafkaCruiseControlConfig config,
     _priorityWeight = config.getDouble(AnalyzerConfig.GOAL_BALANCEDNESS_PRIORITY_WEIGHT_CONFIG);
     _strictnessWeight = config.getDouble(AnalyzerConfig.GOAL_BALANCEDNESS_STRICTNESS_WEIGHT_CONFIG);
     _allowCapacityEstimationOnProposalPrecompute = config.getBoolean(AnalyzerConfig.ALLOW_CAPACITY_ESTIMATION_ON_PROPOSAL_PRECOMPUTE_CONFIG);
-    Map<String, Object> overrideConfigs = Collections.singletonMap(KAFKA_CRUISE_CONTROL_CONFIG_OBJECT_CONFIG, config);
+    Map<String, Object> overrideConfigs = new HashMap<>(2);
```

Comment:
Nit: maybe considering the below approach which is slightly more cohesive. I personally like immutability.

```
    Map<String, Object> overrideConfigs = Collections.unmodifiableMap(new HashMap<String, Object>(2) {{
      put(KAFKA_CRUISE_CONTROL_CONFIG_OBJECT_CONFIG, config);
      put(ADMIN_CLIENT_CONFIG, adminClient);
    }});
```

Reply:
_Double brace initialization_ comes with a significant overhead (please see [efficiency-of-java-double-brace-initialization](https://stackoverflow.com/questions/924285/efficiency-of-java-double-brace-initialization)) and (personally) an unorthodox syntax. Hence, I'd prefer to avoid it.

In this particular use-case, immutability does not provide much benefit as the map is purely generated for the use of `KafkaCruiseControlConfig#getConfiguredInstance` and not for any subsequent use.